task-driver/integration/tests/update_wallet.rs

```diff
@@ -52,11 +53,17 @@ lazy_static! {
 
 /// Perform a wallet update task and verify that it succeeds
 pub(crate) async fn execute_wallet_update(
-    old_wallet: Wallet,
+    mut old_wallet: Wallet,
     new_wallet: Wallet,
     transfer: Option<ExternalTransfer>,
     test_args: IntegrationTestArgs,
-) -> Result<()> {
+) -> Result<Wallet> {
+    // Make sure the Merkle proof is present
+    if old_wallet.merkle_proof.is_none() {
```

Comment:
it feels like `old_wallet.merkle_proof` should never be `None`, this would just mean the wallet wasn't properly allocated in the darkpool, right?

Reply:
Not really, this is just a convenient safety check on the tests. The tests are more brittle the more state we expect them to compose.